src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -245,4 +247,33 @@ private Object resolveConstructor(Class<?> beanClass) {
         }
         throw new BeanFactoryException("Unable to resolve dependencies for class: " + beanClass.getName());
     }
+
+    private Object getBeanDependency(Type dependencyType) {
+        Object dependency;
+        if (dependencyType instanceof ParameterizedType
+            && ((ParameterizedType) dependencyType).getRawType().equals(List.class)) {
+            dependency = getCollectionDependency(dependencyType, 0).toList();
+        } else if (dependencyType instanceof ParameterizedType
+            && ((ParameterizedType) dependencyType).getRawType().equals(Set.class)) {
+            dependency = getCollectionDependency(dependencyType, 0).collect(Collectors.toSet());
+        } else if (dependencyType instanceof ParameterizedType
+            && ((ParameterizedType) dependencyType).getRawType().equals(Map.class)) {
+            dependency = getCollectionDependency(dependencyType, 1)
+                .collect(Collectors.toMap(bean -> bean.getClass().getName(), bean -> bean));
+        } else {
+            dependency = singletonBeans.values().stream()
+                .filter(bean -> bean.getClass().getTypeName().equals(dependencyType.getTypeName()))
```

Comment:
Generally it is unsafe to do such comparison `bean.getClass().getTypeName().equals(dependencyType.getTypeName())`, because if the `dependencyType` is an instance of `ParameterizedType`, `getTypeName` for it returns something like `package.beanName<genericParamClass>`. On the other hand, `bean.getClass().getTypeName()` returns `package.beanName`.

Currently, it won't cause any issues but it is better to adjust it to be future-proof. Simply reuse `rawType` from the previous comment: `.filter(bean -> bean.getClass().equals(rawType))`.

Reply:
Changed